cmd/webhook/main.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	log "github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/cmd/webhook/admission"
+	"golang.org/x/net/context"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+const (
+	defaultHTTPSAddress = ":9443"
+	defaultHTTPAddress  = ":9080"
+)
+
+type config struct {
+	debug    bool
+	certFile string
+	keyFile  string
+	address  string
+}
+
+func (c *config) parse() {
+	kingpin.Flag("debug", "Enable debug logging").BoolVar(&c.debug)
+	kingpin.Flag("tls-cert-file", "File containing the certificate for HTTPS").Envar("CERT_FILE").StringVar(&c.certFile)
+	kingpin.Flag("tls-key-file", "File containing the private key for HTTPS").Envar("KEY_FILE").StringVar(&c.keyFile)
+	kingpin.Flag("address", "The address to listen on").Default(defaultHTTPSAddress).StringVar(&c.address)
+
+	kingpin.Parse()
+
+	if (c.certFile != "" || c.keyFile != "") && !(c.certFile != "" && c.keyFile != "") {
+		log.Fatal("config parse error: both of TLS cert & key must be provided or neither (for testing )")
+		return
+	}
+
+	// support non-HTTPS for local testing
+	if (c.certFile == "" && c.keyFile == "") && c.address == defaultHTTPSAddress {
+		c.address = defaultHTTPAddress
+	}
+
+	if c.debug {
+		log.SetLevel(log.DebugLevel)
+	}
+}
+
+func main() {
+	var cfg config
+	cfg.parse()
+
+	rgAdmitter := admission.RouteGroupAdmitter{}
+	handler := http.NewServeMux()
+	handler.Handle("/routegroups", admission.Handler(rgAdmitter))
+	handler.Handle("/metrics", promhttp.Handler())
+	handler.HandleFunc("/healthz", healthCheck)
+
+	// One can use generate_cert.go in https://golang.org/pkg/crypto/tls
+	// to generate cert.pem and key.pem.
+	serve(cfg.address, cfg.certFile, cfg.keyFile, handler)
+}
+
+func healthCheck(writer http.ResponseWriter, _ *http.Request) {
+	writer.WriteHeader(http.StatusOK)
+	if _, err := writer.Write([]byte("ok")); err != nil {
+		log.Errorf("failed to write health check: %v", err)
+	}
+
+}
+
+func serve(address, certFile, keyFile string, handler http.Handler) {
```

Comment:
since we already have the config type, why just passing a config object to serve instead of the address, certFile and keyFile parameters? It would be easier and nicer to extend it in the future.